sympy/polys/matrices/domainmatrix.py

```diff
@@ -148,6 +156,31 @@ def mul(A, b):
     def matmul(A, B):
         return A.from_rep(A.rep.matmul(B.rep))
 
+    def scalarmul(A, lamda):
+        A, lamda = A.unify(lamda)
+        if lamda.element == lamda.domain.zero:
+            return DomainScalar(A.domain.zero, A.domain)
+        if lamda.element == lamda.domain.one:
+            return A
+
+        return A.from_rep(A.rep.scalarmul(lamda.element))
+
+    def __truediv__(A, lamda):
+        """ Method for Scalar Divison"""
+        if not isinstance(lamda, DomainScalar):
+            if isinstance(lamda, DomainMatrix) or isinstance(lamda, DDM) or isinstance(lamda, SDM):
+                raise TypeError("unsupported operand type(s) for /:  'DomainMatrix' and '%s'" % type(lamda).__name__)
+
+            lamda = DomainScalar.from_sympy(lamda)
```

Comment:
We should not use `from_sympy` here.

Reply:
This felt like a necessity in order to keep `DomainMatrix.scalarmul` free from all kinds of checks and conversions. Otherwise, we have to use it in `DomainMatrix.scalarmul`.
Kindly share your opinion.
Thanks.